Userland/Libraries/LibUnicode/IDNA.cpp

```diff
@@ -0,0 +1,257 @@
+/*
+ * Copyright (c) 2023, Simon Wanner <simon@skyrising.xyz>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/Error.h>
+#include <AK/String.h>
+#include <LibUnicode/IDNA.h>
+#include <LibUnicode/Normalize.h>
+#include <LibUnicode/Punycode.h>
+
+#if ENABLE_UNICODE_DATA
+#    include <LibUnicode/IDNAData.h>
+#    include <LibUnicode/UnicodeData.h>
+#else
+namespace Unicode {
+
+enum class GeneralCategory : u8 {
+    Mark
+};
+
+bool code_point_has_general_category(u32, Unicode::GeneralCategory)
+{
+    return false;
+}
+
+}
```

Comment:
Instead of this, in all other places we do a static general category lookup once at runtime, and handle the case where that lookup fails (i.e. UCD data is disabled).

So in `is_valid_label`, something like:
```c++
static auto general_category_mark = general_category_from_string("Mark"sv);
if (!general_category_mark.has_value()) {
	// Do whatever makes most sense for this library
}
```